enterprise/cmd/worker/internal/embeddings/repo/handler.go

```
@@ -64,18 +65,9 @@ func (h *handler) Handle(ctx context.Context, logger log.Logger, record *repoemb
 	// job for this repo. If we can find one, we'll attempt a delta index, otherwise
 	// we fall back to a full index.
 	var lastSuccessfulJobRevision api.CommitID
+	var previousEmbeddingsIndex *embeddings.RepoEmbeddingIndex
 	if conf.Get().Embeddings.Incremental {
-		lastSuccessfulJob, err := h.repoEmbeddingJobsStore.GetLastCompletedRepoEmbeddingJob(ctx, record.RepoID)
-		if err != nil {
-			logger.Info("no previous successful embeddings job found. Falling back to full index")
-		} else {
-			lastSuccessfulJobRevision = lastSuccessfulJob.Revision
-			logger.Info(
-				"found previous successful embeddings job. Attempting delta index",
-				log.String("old revision", string(lastSuccessfulJobRevision)),
-				log.String("new revision", string(record.Revision)),
-			)
-		}
+		lastSuccessfulJobRevision, previousEmbeddingsIndex = h.getPreviousEmbeddingIndex(ctx, logger, repo)
```

nit: This is very implicit. Maybe `getPreviousEmbeddingIndex` should return an error and not take a logger. Then we simply log the error and set previousEmbeddingsIndex explicitly to "".  Feel free to ignore this.

---

I wasn't able to refactor this in a nice way, since we need to distinguish between an "info" case (no previous job for that repo) and "error" case (we failed to download index).